### PR TITLE
refactor: update email content provider to use caseData

### PR DESCRIPTION
**Pull request by joez-f**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/FPLA-1177

### Change description ###

update AbstractEmailContentProvider to extract information from caseData rather than from caseDetails, also move getFirstRespondentSurname() method from EmailNotificationHelper to a new more appropriate helper class.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/utils/EmailNotificationHelper.java [hunk 1]
```diff
@@ -27,7 +24,7 @@ private EmailNotificationHelper() {
     }
 
     public static String buildSubjectLine(final CaseData caseData) {
-        final String lastName = getFirstRespondentLastName(caseData);
+        final String lastName = getFirstRespondentLastName(caseData.getRespondents1());
```
Comment (tomaszpowroznik) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/utils/EmailNotificationHelper.java]: respondentLastName maybe

File: service/src/main/java/uk/gov/hmcts/reform/fpl/utils/PeopleInCaseHelper.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import static org.springframework.util.CollectionUtils.isEmpty;
+
+public class PeopleInCaseHelper {
+
+    private PeopleInCaseHelper() {
+        // NO-OP
+    }
+
+    public static String getFirstRespondentLastName(List<Element<Respondent>> respondents) {
+        return isEmpty(respondents) ? "" : respondents
```
Comment (tomaszpowroznik) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/utils/PeopleInCaseHelper.java]: can utilise ElementUtils 

ElementUtils.unwrapElements(respondents)
            .stream()
            .filter(Objects::nonNull)
            .findFirst()
            .map(Respondent::getParty)
            .map(RespondentParty::getLastName)
            .orElse("");

File: service/src/test/java/uk/gov/hmcts/reform/fpl/utils/PeopleInCaseHelperTest.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+        String respondentName = getFirstRespondentLastName(createRespondents());
+        assertThat(respondentName).isEqualTo("Jones");
+    }
+
+    @Test
+    void shouldReturnEmptyStringWhenNoRespondents() {
+
+        String respondentName = getFirstRespondentLastName(null);
+        assertThat(respondentName).isEqualTo("");
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/utils/PeopleInCaseHelperTest.java]: what about .isEmpty()

File: service/src/test/java/uk/gov/hmcts/reform/fpl/utils/PeopleInCaseHelperTest.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    @Test
+    void shouldReturnEmptyStringWhenRespondentWithNoPartyPresent() {
+
+        String respondentName = getFirstRespondentLastName(wrapElements(Respondent.builder().build()));
+        assertThat(respondentName).isEqualTo("");
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/utils/PeopleInCaseHelperTest.java]: same here

File: service/src/test/java/uk/gov/hmcts/reform/fpl/utils/PeopleInCaseHelperTest.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    @Test
+    void shouldReturnEmptyStringWhenRespondentWithNoNamePresent() {
+        List<Element<Respondent>> respondents = wrapElements(Respondent.builder()
+            .party(RespondentParty.builder().build())
+            .build());
+
+        String respondentName = getFirstRespondentLastName(respondents);
+        assertThat(respondentName).isEqualTo("");
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/utils/PeopleInCaseHelperTest.java]: same here

File: service/src/main/java/uk/gov/hmcts/reform/fpl/utils/PeopleInCaseHelper.java (hunk 2 above)
Comment (joez-f) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/utils/PeopleInCaseHelper.java]: thanks!

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/AbstractEmailContentProvider.java [hunk 6]
```diff
@@ -80,38 +81,38 @@ private String getHearingBooking(CaseData data) {
...
         return ordersAndDirectionsBuilder.build();
     }
 
     @SuppressWarnings("unchecked")
-    private void appendOrders(Map<String, Object> orders, ImmutableList.Builder<String> builder) {
-        Optional.ofNullable(orders.get("orderType")).ifPresent(orderTypes -> {
-            for (String typeString : (List<String>) orderTypes) {
-                builder.add(OrderType.valueOf(typeString).getLabel());
+    private void appendOrders(Orders orders, ImmutableList.Builder<String> builder) {
+        Optional.ofNullable(orders.getOrderType()).ifPresent(orderTypes -> {
```
[the remark this answers is not in the document]
Comment (joez-f) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/AbstractEmailContentProvider.java]: concise, I like it 👍

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/AbstractEmailContentProvider.java [hunk 7]
```diff
@@ -67,8 +68,8 @@ protected AbstractEmailContentProvider(String uiBaseUrl,
                 .getParty()
                 .getLastName()) + ",")
             .put("hearingDate", getHearingBooking(caseData))
-            .put("reference", String.valueOf(caseDetails.getId()))
-            .put("caseUrl", uiBaseUrl + "/case/" + JURISDICTION + "/" + CASE_TYPE + "/" + caseDetails.getId());
+            .put("reference", String.valueOf(caseDetailsId))
+            .put("caseUrl", uiBaseUrl + "/case/" + JURISDICTION + "/" + CASE_TYPE + "/" + caseDetailsId);
```
[the remark this answers is not in the document]
Comment (joez-f) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/AbstractEmailContentProvider.java]: Yeah, I'll do it

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/Orders.java [hunk 8]
```diff
@@ -16,6 +18,7 @@
     @NotNull(message = "Select at least one type of order")
     @Size(min = 1, message = "Select at least one type of order")
     private final List<OrderType> orderType;
+    private final List<EmergencyProtectionOrdersType> emergencyProtectionOrders;
     private final String directions;
-    private final List<String> emergencyProtectionOrderDirections;
```
Comment (joez-f) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/model/Orders.java]: @akomisarek does this break things? here's the ccd
```
  {
    "LiveFrom": "01/01/2017",
    "ID": "Orders",
    "ListElementCode": "emergencyProtectionOrderDirections",
    "FieldType": "MultiSelectList",
    "FieldTypeParameter": "EmergencyProtectionOrderDirectionType",
    "ElementLabel": "Do you need any of these directions?",
    "FieldShowCondition": "orderType CONTAINS \"EMERGENCY_PROTECTION_ORDER\"",
    "SecurityClassification": "Public"
  },
```

**Comment by joez-f** [comment]
@geek-soft PR for the code review comments you left on the application notification story, would be good if you can take a look
also @tomaszpowroznik can you re-review? I made couple of changes

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/AbstractEmailContentProvider.java [hunk 9]
```diff
@@ -36,13 +39,11 @@ protected AbstractEmailContentProvider(String uiBaseUrl,
...
-    ImmutableMap.Builder<String, Object> getCasePersonalisationBuilder(CaseDetails caseDetails, CaseData caseData) {
-        List<String> ordersAndDirections = buildOrdersAndDirections(
-            (Map<String, Object>) caseDetails.getData().get("orders"));
+    ImmutableMap.Builder<String, Object> getCasePersonalisationBuilder(Long caseId, CaseData caseData) {
+        List<String> ordersAndDirections = buildOrdersAndDirections(caseData.getOrders());
 
-        Optional<String> timeFrame = Optional.ofNullable((Map<String, Object>) caseDetails.getData().get("hearing"))
-            .map(hearing -> (String) hearing.get("timeFrame"));
+        Optional<String> timeFrame = Optional.ofNullable(caseData.getHearing()).map(
+            Hearing::getTimeFrame).filter(StringUtils::isNotBlank);
```
Comment (tomaszpowroznik) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/AbstractEmailContentProvider.java]: minor, could be formatted as:
Optional.ofNullable(caseData.getHearing())
            .map(Hearing::getTimeFrame)
            .filter(StringUtils::isNotBlank)

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/AbstractEmailContentProvider.java [hunk 10]
```diff
@@ -80,7 +81,12 @@ private String getHearingBooking(CaseData data) {
         return "";
     }
 
-    private List<String> buildOrdersAndDirections(Map<String, Object> optionalOrders) {
+    private String formatCaseURL(Long caseId) {
+        return String.format("%s/case/%s/%s/%s", uiBaseUrl, JURISDICTION, CASE_TYPE, caseId);
+
```
Comment (tomaszpowroznik) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/AbstractEmailContentProvider.java]: minor: not needed empty line

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/AbstractEmailContentProvider.java [hunk 11]
```diff
@@ -91,28 +97,21 @@ private String getHearingBooking(CaseData data) {
...
 
-    @SuppressWarnings("unchecked")
-    private void appendOrders(Map<String, Object> orders, ImmutableList.Builder<String> builder) {
-        Optional.ofNullable(orders.get("orderType")).ifPresent(orderTypes -> {
-            for (String typeString : (List<String>) orderTypes) {
-                builder.add(OrderType.valueOf(typeString).getLabel());
-            }
-        });
+    private void appendOrders(Orders orders, ImmutableList.Builder<String> builder) {
+        defaultIfNull(orders.getOrderType(), Collections.<OrderType>emptyList()).stream()
```
Comment (swalker125) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/AbstractEmailContentProvider.java]:
```suggestion
        defaultIfNull(orders.getOrderType(), emptyList()).stream()
```

I don't think you need to be verbose with type here

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/Orders.java (hunk 8 above)
Comment (swalker125) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/Orders.java]: if your enum maps to the same strings, no it won't break

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/AbstractEmailContentProvider.java (hunk 11 above)
Comment (tomaszpowroznik) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/AbstractEmailContentProvider.java]: @swalker125 it wont compile without explicit type information, unless you do
      List<OrderType> types=  defaultIfNull(orders.getOrderType(), emptyList());

        types.stream()
            .map(OrderType::getLabel)
            .forEach(builder::add);

**Pull request merged**